app/Get.hs

```diff
@@ -0,0 +1,75 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Get where
+
+import Control.Monad.IO.Class
+import Data.Text (Text, pack, unpack)
+import Data.Aeson
+import Network.HTTP.Req
+import Fragnix.Slice (Slice, writeSlice, sliceNestedPath, moduleNameReference)
+import Fragnix.Core.Slice (uses, reference, Reference(OtherSlice, Builtin))
+import Fragnix.Environment (writeSymbols)
+import Fragnix.Paths (slicesPath, environmentPath)
+import Language.Haskell.Names (Symbol, symbolModule)
+import Language.Haskell.Exts (ModuleName (..), prettyPrint)
+import Data.Maybe (isJust)
+import Text.Read (readMaybe)
+import Control.Monad (forM_, filterM)
+import System.Directory (createDirectoryIfMissing, doesFileExist)
+import System.FilePath ((</>))
+import Utils (IDType (SliceID, EnvID), sliceRequest, envRequest)
+
+get :: IDType -> String -> Bool -> IO ()
+get SliceID = fetchSlice
+get EnvID = fetchEnv
+
+fetchSlice :: String -> Bool -> IO ()
+fetchSlice id nodeps = do
+  sliceExists <- doesSliceExist id
+  if sliceExists
+    then putStrLn $ "Slice " ++ id ++ " already fetched."
+    else do
+      putStrLn $ "Fetching slice " ++ id
+      r <- sliceRequest id
+      let slice = responseBody r :: Slice
+      writeSlice slicesPath slice
+      if nodeps
+        then 
+          return ()
+        else do
+          let references = fmap reference $ uses slice
+          let slicesToDownload = [sliceID | (OtherSlice sliceID) <- references]
+          forM_ slicesToDownload (\slice -> fetchSlice (unpack slice) False)
+
+fetchEnv :: String -> Bool -> IO ()
+fetchEnv id nodeps = do
+  let modulePath = environmentPath </> id
+  envExists <- doesFileExist modulePath
+  if envExists
+    then putStrLn $ "Environment " ++ id ++ " already fetched."
+    else do
+      putStrLn $ "Fetching environment " ++ id
```

Comment:
Similar bug possible as above, when environment exists but the dependencies don't yet exist. We then fail to get the dependencies.